src/resultset/resultset.c

```diff
@@ -127,9 +127,13 @@ ResultSet *NewResultSet(RedisModuleCtx *ctx, bool compact) {
 	return set;
 }
 
-void ResultSet_SetColumns(ResultSet *set, const char **columns) {
-	assert(set && columns);
-	set->columns = columns;
+void ResultSet_SetColumns(const char **columns) {
+	ResultSet *set = QueryCtx_GetResultSet();
```

Comment:
I don't like this design where we call `ResultSet_*` functions without passing a result-set object,
please revert.